### PR TITLE
feat: Add option to set http attributes when creating an instance of Client

### PR DESCRIPTION
**Pull request by jrectenwald**
This PR is addressing issue [#13](https://github.com/sendgrid/ruby-http-client/issues/13)

**The Problem**
The issue requests the ability to override the Net::HTTP defaults for open and read timeouts.

**What I did**
I added a parameter called http_options to the Sendgrid::Client class. This parameter should be a hash with key/value pairs corresponding with attributes for NET:HTTP as described [here](https://ruby-doc.org/stdlib-2.4.1/libdoc/net/http/rdoc/Net/HTTP.html). I added this in the test file and added this to the example file as well.

### CODE REVIEW

File: test/test_ruby_http_client.rb [hunk 1]
```diff
@@ -31,9 +31,11 @@ def setup
         ')
     @host = 'http://localhost:4010'
     @version = 'v3'
```
Comment (mbernier) [inline test/test_ruby_http_client.rb:34]: @jrectenwald Would you be OK with adding a duplicate test to this one, that doesn't pass http_options, so we can prove in the tests that it works both ways?

**Review by mbernier — CHANGES_REQUESTED** [review]

File: test/test_ruby_http_client.rb (hunk 1 above)
Comment (jrectenwald) [reply test/test_ruby_http_client.rb:34]: @mbernier How's that?

**Comment by thinkingserious** [comment]
Hello @jrectenwald,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/ruby-http-client?pullRequest=20) <br/>All committers have signed the CLA.

**Pull request merged**